misc/userscripts/password_fill

```diff
@@ -209,6 +210,81 @@ choose_entry_zenity_radio() {
     choose_entry_menu || true
 }
 
+# Default backend selection.
+backend="pass"
+
+# =======================================================
+# backend: LastPass. 
+# The LastPass backend depnds only on a system installation of lastpass-cli.
+# Users must define their lastpass username in password_fill_rc and set 
+# the backend variable to "last_pass"
+# e.g.
+#
+# #!/usr/bin/env bash
+# username=lp-username@email.com
+# backend=last_pass
+#
+# Temporary files are created that store account user name along
+# with the lastpass id for a given address. Users can redefine where the
+# temporary files will be saved with LP_CACHE_PREFIX the default is as such.
+#
+# No password data is ever written to a file with this method.
+
+LP_CACHE_PREFIX=$HOME/.config/qutebrowser/id_cache
+
+last_pass_backend() {
+    init() {
+      [ ! -d $LP_CACHE_PREFIX ] && mkdir $LP_CACHE_PREFIX
+      [ ! -d $LP_CACHE_PREFIX ] && die "Cannot make LP_CACHE_PREFIX dir : $LP_CACHE_PREFIX"
+
+      PREFIX=$LP_CACHE_PREFIX
+      set +e
+
+      # Remove any old LP temporary id files.
+      for f in $PREFIX/; do
+        [ -f $f ] && rm $f 
+      done
+
+      # Check to see if the system is logged into LP else prompt
+      # the user with the LP login.
+      lpass ls
+      if ! [ $? -eq 0 ]; then
+        lpass login --trust $username
+        lpass ls
+        if [ $? -eq 0 ]; then
+          msg "Lastpass successful login."
+        else
+          die "Lastpass failed to initialize correctly."
+        fi
+      fi
+      set -e
+    }
+
+    query_entries() {
+        local url="$1"
+
+        set +e
+        lpass show $url
+        if [ $? -eq 0 ]; then
+            # Save files as usernames w/ corresponding LP id token.
+            for id in $(lpass show --id -x $url); do
+                echo "$id" > $PREFIX/$(lpass show --username -x $id)
+                files+=("$(lpass show --username -x $id)")
+            done
+        fi
+        set -e
+    }
+
+    open_entry() {
+        # Use the id tokens to pull password form the correct LP profile.
+        username="${1#* }"
+        id=$(cat $PREFIX/$username)
+        password=$(lpass show --password -x $id)
```

Comment:
I don't get why you need the the temporary file at all. The file content contains the output of the command
```
lpass show --username -x $id
```
and you extend the `files` array by the output of the exactly same command, so `$1` which is passed to `open_entry` is again the output of this command, so even knowning nothing about the output of `lpass show` I conjecture that you can implement the `open_entry()` function as follows:
```bash
open_entry() {
username="${1#* }"
password=$(lpass show --password -x $1)
}
```
Making it entirely unnecessary to write the temporary files in the `query_entries()` function. Of course, one should add proper quoting around `"$1"`, but this requires knowledge about the format of the output of `lpass show`.